qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/numpy_minimum_eigensolver_factory.py

```diff
@@ -87,9 +107,13 @@ def get_solver(
         filter_criterion = self._filter_criterion
         if not filter_criterion and self._use_default_filter_criterion:
             filter_criterion = problem.get_default_filter_criterion()
-
-        npme = NumPyMinimumEigensolver(filter_criterion=filter_criterion)
-        return npme
+            self._minimum_eigensolver.filter_criterion = filter_criterion
+        return self._minimum_eigensolver
 
     def supports_aux_operators(self):
         return NumPyMinimumEigensolver.supports_aux_operators()
+
+    @property
+    def minimum_eigensolver(self):
```

Comment:
I guess this should be typehinted for a return of NumPyMinimumEigensolver since we want to allow the user full access to the specific type and its properties here right . I saw the same method ion the VQE factories and was initially thinking perhaps this could be a method on the base class. But we want the user of VQE factories to be able to get to the VQE and alter things like include_custom etc., and not work with it through the common MinimumEigenSolver interface that would presumably be the only type that would make sense for such a function in the base class to say it returns.

@mrossinek With this change we are doing there is subtle behavior change in that before a factory could emit number of solvers and they were independent. Now its more one factory one solver. I think its ok, its more where we want to be in having the user have direct access to the algorithm and the user changing it directly via its properties rather than us having all the indirects here.